src/db.rs

```diff
@@ -1152,6 +1152,48 @@ impl DB {
             ))
         })
     }
+
+    pub fn property_value(&self, name: &str) -> Option<String> {
```

Comment:
That sounds reasonable to me, @iSynaptic. In that case I think it make more sense to return an error when `value.is_null()` as well though, since that means the property doesn't exist: https://github.com/facebook/rocksdb/blob/a62c6626e0ae27390013abcc94b42cecd6eb4497/include/rocksdb/c.h#L330-L333

Not sure what to do for `property_int_value` — just return an `Error::new(format!('failed to parse value: {}))'` or something more specific? It's a shame the RocksDB C-bindings don't expose `GetIntProperty`.

What do you think?

Reply:
And I guess if it returned an error for missing properties, `Result<String, Error>` would be a better type, since there wouldn't be a `None` case.

I'm not completely confident that an error for a missing property is the right thing to do here though, but I feel like it signifies a mistake a little more than, say, a missing hash map key, since the set of properties are essentially fixed. I don't write much Rust anymore though, so definitely open to feedback.